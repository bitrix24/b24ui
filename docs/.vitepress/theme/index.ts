/**
 * This is custom theme
 * We change some css
 * @see node_modules/vitepress/dist/client/theme-default/index.js
 */
import { ref, watch, toValue } from 'vue'
import { createRouter, createMemoryHistory } from 'vue-router'
import bitrix24UIPlugin from '@bitrix24/b24ui-nuxt/vue-plugin'
import Layout from './components/ui/Layout.vue'
import ComponentShowExample from './components/ui/ComponentShowExample.vue'
import ComponentProps from './components/ui/ComponentProps.vue'
import ComponentSlots from './components/ui/ComponentSlots.vue'
import ComponentEmits from './components/ui/ComponentEmits.vue'
import Description from './components/ui/Description.vue'
import Placeholder from './components/ui/Placeholder.vue'
import VPBadge from 'vitepress/dist/client/theme-default/components/VPBadge.vue'
import 'vitepress/dist/client/theme-default/styles/fonts.css'
import './tailwind.css'
// region ory theme ////
import './styles/components/custom-block.css'
import './styles/components/vp-code.css'
import './styles/components/vp-code-group.css'
import './styles/components/vp-doc.css'
import './styles/components/vp-sponsor.css'
// endregion ////
import './theme.css'

globalThis.useFetch = async (url: string, options: RequestInit & { params?: any, transform?: (data) => any } = {}) => {
  const data = ref()
  const status = ref('idle')
  async function _fetch() {
    status.value = 'loading'
    try {
      const params = options.params
        ? Object.fromEntries(
            Object.entries(options.params).map(([key, value]) => [key, toValue(value)])
          )
        : {}

      const queryString = new URLSearchParams(params as any).toString()
      const fullUrl = queryString ? `${url}?${queryString}` : url

      data.value = await fetch(fullUrl, options).then(r => r.json()).then(r => options.transform ? options.transform(r) : r)
      status.value = 'success'
    } catch (error) {
      console.error(error)
      status.value = 'error'
    }
  }
  _fetch()

  if (options?.params?.q) {
    watch(
      options.params.q,
      () => {
        _fetch()
      },
      { deep: true }
    )
  }

  return Promise.resolve({
    data,
    status
  })
}

export default {
  Layout,
  enhanceApp: async ({ app }) => {
    // region components ////
    app.component('Badge', VPBadge)
      .component('Description', Description)
      .component('ComponentShowExample', ComponentShowExample)
      .component('ComponentProps', ComponentProps)
      .component('ComponentSlots', ComponentSlots)
      .component('ComponentEmits', ComponentEmits)
      .component('Placeholder', Placeholder)
    // endregion ////

    if (!import.meta.env.SSR) {
      const router = createRouter({
        routes: [
          // { path: '', component: app }
        ],
        history: createMemoryHistory()
        // history: createWebHistory()
      })
      app.use(router)

      app.use(bitrix24UIPlugin)
    }
  }
}
