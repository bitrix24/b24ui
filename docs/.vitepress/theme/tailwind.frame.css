/*
@plugin '@bitrix24/b24style';
@plugin "@bitrix24/b24ui-nuxt";
*/
@import "tailwindcss" source("../../");
@import "@bitrix24/b24ui-nuxt";
/*
*/



