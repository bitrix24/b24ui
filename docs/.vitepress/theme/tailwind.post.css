@import "tailwindcss" source("../../../docs");
@import "@bitrix24/b24ui-nuxt";

/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}
