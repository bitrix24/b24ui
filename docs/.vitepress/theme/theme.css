/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
  --sh-scrollbar-thumb: #3c3c43;
  --sh-scrollbar-background: #ebebef;
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
  --vp-code-block-bg: #1a2234;

  --sh-scrollbar-thumb: #bdc1c6;
  --sh-scrollbar-background: #3c3c43;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

iframe
{
  min-width: 100%;
  border: none;
  scrollbar-width: thin;
  background-color: transparent;
}

/** region scrollbar */
* {
  scrollbar-color: var(--sh-scrollbar-thumb) var(--sh-scrollbar-background);
  scrollbar-width: thin;
}

/* Styling the scrollbar in Chrome and Safari */
*::-webkit-scrollbar {
  width: 6px;
}

/* Set thumb color for Chrome and Safari */
*::-webkit-scrollbar-thumb {
  background-color: var(--sh-scrollbar-thumb);
  border-radius: 4px;
}

/* Set track color for Chrome and Safari */
*::-webkit-scrollbar-track {
  background-color: var(--sh-scrollbar-background);
}
/** endregion */
