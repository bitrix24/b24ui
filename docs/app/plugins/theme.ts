export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
  }
})
