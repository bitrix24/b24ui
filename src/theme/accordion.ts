/**
 * Accordion
 * A stacked set of collapsible panels.
 * ---
 * @see Separator.accent
 */

export default {
  slots: {
    root: 'w-full',
    item: 'text-(--b24ui-typography-label-color) border-b border-(--ui-color-divider-vibrant-accent-more) last:border-b-0',
    header: 'flex',
    trigger: [
      'min-w-0 group flex-1 flex items-center gap-1.5',
      'py-[12px]',
      'font-(--ui-font-weight-medium) text-(length:--ui-font-size-sm) leading-[20px]',
      'focus-visible:outline-(--ui-color-accent-soft-element-blue)',
      'cursor-pointer'
    ].join(' '),
    content: 'motion-safe:data-[state=open]:animate-[accordion-down_200ms_ease-out] motion-safe:data-[state=closed]:animate-[accordion-up_200ms_ease-out] overflow-hidden focus:outline-none',
    body: 'text-(length:--ui-font-size-sm) pb-[12px]',
    leadingIcon: 'shrink-0 size-[20px]',
    trailingIcon: 'shrink-0 size-[20px] ms-auto group-data-[state=open]:rotate-180 transition-transform duration-200',
    label: 'text-start break-words'
  },
  variants: {
    disabled: {
      true: {
        trigger: 'cursor-not-allowed opacity-75'
      }
    }
  }
}
