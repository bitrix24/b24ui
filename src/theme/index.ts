export { default as advice } from './advice'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as container } from './container'
export { default as countdown } from './countdown'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as range } from './range'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as switch } from './switch'
export { default as textarea } from './textarea'
export { default as tooltip } from './tooltip'
