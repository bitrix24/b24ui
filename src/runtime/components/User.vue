<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/user'
import type { AvatarProps, ChipProps, LinkProps } from '../types'
import type { ComponentConfig } from '../types/tv'

type User = ComponentConfig<typeof theme, AppConfig, 'user'>

export interface UserProps {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  name?: string
  description?: string
  avatar?: Omit<AvatarProps, 'size'> & { [key: string]: any }
  chip?: boolean | Omit<ChipProps, 'size' | 'inset'>
  /**
   * @defaultValue 'md'
   */
  size?: User['variants']['size']
  /**
   * The orientation of the user.
   * @defaultValue 'horizontal'
   */
  orientation?: User['variants']['orientation']
  to?: LinkProps['to']
  target?: LinkProps['target']
  onClick?: (event: MouseEvent) => void | Promise<void>
  class?: any
  b24ui?: User['slots']
}

export interface UserSlots {
  avatar(props: { b24ui: User['b24ui'] }): any
  name(props?: {}): any
  description(props?: {}): any
  default(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed } from 'vue'
import { Primitive } from 'reka-ui'
import { useAppConfig } from '#imports'
import { tv } from '../utils/tv'
import B24Chip from './Chip.vue'
import B24Avatar from './Avatar.vue'
import B24Link from './Link.vue'

defineOptions({ inheritAttrs: false })

const props = withDefaults(defineProps<UserProps>(), {
  orientation: 'horizontal'
})
const slots = defineSlots<UserSlots>()

const appConfig = useAppConfig() as User['AppConfig']

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.user || {}) })({
  size: props.size,
  orientation: props.orientation,
  to: !!props.to || !!props.onClick
}))

const chipSize = computed<ChipProps['size']>(() => {
  switch (props.size) {
    case '3xs': return 'sm' as ChipProps['size']
    case '2xs': return 'sm' as ChipProps['size']
    case 'xs': return 'sm' as ChipProps['size']
    case 'md': return 'sm' as ChipProps['size']
    case 'lg': return 'md' as ChipProps['size']
    case 'xl': return 'md' as ChipProps['size']
    case '2xl': return 'lg' as ChipProps['size']
    case '3xl': return 'lg' as ChipProps['size']
  }

  return 'sm' as ChipProps['size']
})
</script>

<template>
  <Primitive :as="as" :data-orientation="orientation" :class="b24ui.root({ class: [props.b24ui?.root, props.class] })" @click="onClick">
    <slot name="avatar" :b24ui="b24ui">
      <B24Chip v-if="chip && avatar && !['3xs'].includes(size || '')" inset v-bind="typeof chip === 'object' ? chip : {}" :size="chipSize">
        <B24Avatar :alt="name" v-bind="avatar" :size="size" :class="b24ui.avatar({ class: props.b24ui?.avatar })" />
      </B24Chip>
      <B24Avatar v-else-if="avatar" :alt="name" v-bind="avatar" :size="size" :class="b24ui.avatar({ class: props.b24ui?.avatar })" />
    </slot>

    <div :class="b24ui.wrapper({ class: props.b24ui?.wrapper })">
      <B24Link
        v-if="to"
        :aria-label="name"
        v-bind="{ to, target, ...$attrs }"
        class="focus:outline-none peer"
        tabindex="-1"
        raw
      >
        <span class="absolute inset-0" aria-hidden="true" />
      </B24Link>

      <slot>
        <p v-if="name || !!slots.name" :class="b24ui.name({ class: props.b24ui?.name })">
          <slot name="name">
            {{ name }}
          </slot>
        </p>
        <p v-if="description || !!slots.description" :class="b24ui.description({ class: props.b24ui?.description })">
          <slot name="description">
            {{ description }}
          </slot>
        </p>
      </slot>
    </div>
  </Primitive>
</template>
