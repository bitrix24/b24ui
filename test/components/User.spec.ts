import { describe, it, expect } from 'vitest'
import User from '../../src/runtime/components/User.vue'
import type { UserProps, UserSlots } from '../../src/runtime/components/User.vue'
import ComponentRender from '../component-render'
import theme from '#build/b24ui/user'

describe('User', () => {
  const sizes = Object.keys(theme.variants.size) as any
  const orientations = Object.keys(theme.variants.orientation) as any

  const props = {
    name: 'User name',
    description: 'User description',
    avatar: { src: 'https://github.com/bitrix24.png', alt: 'User alt' }
  }

  it.each([
    // Props
    ['with name', { props: { name: props.name } }],
    ['with description', { props: { name: props.name, description: props.description } }],
    ['with to', { props: { ...props, to: 'https://github.com/bitrix24' } }],
    ['with avatar', { props }],
    ['with chip', { props: { ...props, chip: { color: 'success' } } }],
    ...sizes.map((size: string) => [`with size ${size}`, { props: { ...props, size } }]),
    ...orientations.map((orientation: string) => [`with orientation ${orientation}`, { props: { ...props, orientation } }]),
    ['with as', { props: { ...props, as: 'section' } }],
    ['with class', { props: { ...props, class: 'absolute' } }],
    ['with b24ui', { props: { ...props, b24ui: { name: 'font-bold' } } }],
    // Slots
    ['with avatar slot', { props, slots: { avatar: () => 'Avatar slot' } }],
    ['with name slot', { props, slots: { name: () => 'Name slot' } }],
    ['with description slot', { props, slots: { description: () => 'Description slot' } }],
    ['with default slot', { props, slots: { default: () => 'Default slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: UserProps, slots?: Partial<UserSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, User)
    expect(html).toMatchSnapshot()
  })
})
