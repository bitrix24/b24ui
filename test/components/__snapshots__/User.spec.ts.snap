// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`User > renders with as correctly 1`] = `
"<section data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</section>"
`;

exports[`User > renders with avatar correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with avatar slot correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2">Avatar slot<div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with b24ui correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm) font-bold">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with chip correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2">
  <div class="relative shrink-0 isolate inline-flex items-center justify-center"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span><span class="ui-counter__scope --air font-[family-name:var(--ui-font-family-primary)] select-none min-w-(--ui-counter-size) h-(--ui-counter-size) py-0 px-(--ui-counter-inline-space) inline-flex items-center justify-center bg-(--b24ui-background) rounded-(--ui-counter-current-size) ring-(length:--b24ui-border-width) ring-(--b24ui-border-color) text-center align-middle text-(length:--ui-counter-font-size) text-(--b24ui-color) leading-(--ui-counter-current-size) overflow-hidden z-1 text-nowrap style-filled-success ui-counter-sm font-(--ui-font-weight-regular) top-0 right-0 absolute -translate-y-1/3 translate-x-1/3 transform"><span></span>
    <!----></span>
  </div>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with class correctly 1`] = `
"<div data-orientation="horizontal" class="group/user flex items-center gap-2 absolute"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with default slot correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->Default slot
  </div>
</div>"
`;

exports[`User > renders with description correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2">
  <!--v-if-->
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with description slot correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">Description slot</p>
  </div>
</div>"
`;

exports[`User > renders with name correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2">
  <!--v-if-->
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <!--v-if-->
  </div>
</div>"
`;

exports[`User > renders with name slot correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">Name slot</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with orientation horizontal correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with orientation vertical correctly 1`] = `
"<div data-orientation="vertical" class="relative group/user flex flex-col gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size 2xl correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-3"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring-(--ui-color-base-7) ring-2 size-[60px] text-(length:--ui-font-size-5xl)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="60" height="60" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-md)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-md)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size 2xs correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-1.5"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[20px] text-(length:--ui-font-size-4xs)/(--ui-font-line-height-reset) font-(--ui-font-weight-regular) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="14" height="14" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="flex items-center gap-1.5">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-xs)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size 3xl correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-3"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring-(--ui-color-base-7) ring-2 size-[94px] text-[34px]/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="94" height="94" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-lg)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-md)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size 3xs correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-1"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[10px] text-4xs font-(--ui-font-weight-regular) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="10" height="10" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="flex items-center gap-1">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-xs)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size lg correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2.5"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[42px] text-(length:--ui-font-size-2xl)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="42" height="42" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-sm)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size md correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size sm correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[28px] text-(length:--ui-font-size-xs)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="22" height="22" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-xs)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size xl correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2.5"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring-(--ui-color-base-7) ring-2 size-[48px] text-(length:--ui-font-size-2xl)/(--ui-font-line-height-reset) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="48" height="48" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-md)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-sm)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with size xs correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-1.5"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[24px] text-(length:--ui-font-size-3xs)/(--ui-font-line-height-reset) font-(--ui-font-weight-regular) shrink-0"><img src="https://github.com/bitrix24.png" alt="User alt" width="18" height="18" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class="flex items-center gap-1.5">
    <!--v-if-->
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) text-(length:--ui-font-size-xs)">User name</p>
    <p class="text-(--b24ui-typography-description-color) text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;

exports[`User > renders with to correctly 1`] = `
"<div data-orientation="horizontal" class="relative group/user flex items-center gap-2"><span class="air-secondary-accent inline-flex items-center justify-center select-none rounded-full align-middle bg-(--ui-color-base-8) ring ring-(--ui-color-base-7) size-[32px] text-(length:--ui-font-size-sm)/(--ui-font-line-height-reset) shrink-0 transform transition-transform duration-200 group-hover/user:scale-115"><img src="https://github.com/bitrix24.png" alt="User alt" width="32" height="32" class="h-full w-full rounded-[inherit] object-cover"></span>
  <div class=""><a href="https://github.com/bitrix24" tabindex="-1" rel="noopener noreferrer" aria-label="User name" class="focus:outline-none peer"><span class="absolute inset-0" aria-hidden="true"></span></a>
    <p class="font-(--ui-font-weight-medium) text-(--b24ui-typography-label-color) peer-hover:text-(--ui-color-design-selection-content) transition-colors text-(length:--ui-font-size-sm)">User name</p>
    <p class="text-(--b24ui-typography-description-color) peer-hover:text-(--ui-color-design-selection-content) transition-colors text-(length:--ui-font-size-xs)">User description</p>
  </div>
</div>"
`;
